src/components/carousel/index.tsx

```diff
@@ -116,6 +121,29 @@ const Carousel: FC<Props> = (props) => {
     return () => document.removeEventListener('keydown', keydownListener);
   }, [fullScreen, scrollNext, scrollPrev]);
 
+  const current = mainCarousel?.selectedScrollSnap();
+  const previous = mainCarousel?.previousScrollSnap();
+
+  useEffect(() => {
```

Comment:
I think you would want to use the `onSelect` event from Embla. It should already be registered to select the current slide index. This runs on every slide change and on mount:
https://www.embla-carousel.com/api/events/#select

Reply:
here: https://github.com/smg-automotive/components-pkg/blob/main/src/components/carousel/index.tsx#L98

Reply:
Fair, that makes sense. I can move the logic there